continuous_integration/recipe/meta.yaml

```diff
@@ -42,15 +45,15 @@ requirements:
     - pygments >=2.7.1
     - tabulate
 
-test:
-  imports:
-    - dask_sql
-  commands:
-    - pip check
-    - dask-sql-server --help
-    - dask-sql --help
-  requires:
-    - pip
+test:                                        # [build_platform == target_platform]
+  imports:                                   # [build_platform == target_platform]
+    - dask_sql                               # [build_platform == target_platform]
+  commands:                                  # [build_platform == target_platform]
+    - pip check                              # [build_platform == target_platform]
+    - dask-sql-server --help                 # [build_platform == target_platform]
+    - dask-sql --help                        # [build_platform == target_platform]
+  requires:                                  # [build_platform == target_platform]
+    - pip                                    # [build_platform == target_platform]
```

Comment:
When we run tests on the `linux-aarch64` packages, we get the following error:

```
/usr/share/miniconda3/envs/test/conda-bld/dask-sql_1683837509285/test_tmp/conda_test_runner.sh: line 3: /usr/share/miniconda3/envs/test/conda-bld/dask-sql_1683837509285/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/python: cannot execute binary file: Exec format error
```

Which I'm assuming is because we're attempting to use ARM python on an x86 runner? Not sure if there's a way to unblock tests outside of outright skipping tests for the `linux-aarch64` packages (@jakirkham not sure if you have any thoughts here)

Reply:
Yeah we can't run tests on a different architecture from the binaries we have

Instead of using selectors here, would suggest making the following change to the GHA workflow

```yaml
      - name: Build conda packages
        run: |
          # suffix for nightly package versions
          export VERSION_SUFFIX=a`date +%y%m%d`
          conda mambabuild continuous_integration/recipe \
                           --no-anaconda-upload \
                           --no-test
                           --output-folder .

      - name: Test conda package
        run: |
          conda mambabuild --test linux-64/*.tar.bz2
```

This way one can still run the tests on the package locally (if they have the right architecture) and we skip them on CI when we don't have that architecture